src/Visualization/Python/PlotSizeControl.py

```diff
@@ -0,0 +1,361 @@
+            times = np.concatenate([times, diagnostics_data[:, 0]])
+            control_error = np.concatenate(
+                [control_error, diagnostics_data[:, 1]]
+            )
+            state = np.concatenate([state, diagnostics_data[:, 2]])
+            function_of_time = np.concatenate(
+                [function_of_time, diagnostics_data[:, 4]]
+            )
+            dt_function_of_time = np.concatenate(
+                [dt_function_of_time, diagnostics_data[:, 5]]
+            )
+            horizon_coef_00 = np.concatenate(
+                [horizon_coef_00, diagnostics_data[:, 6]]
+            )
+            avg_dt_horizon_coef_00 = np.concatenate(
+                [avg_dt_horizon_coef_00, diagnostics_data[:, 7]]
+            )
+            raw_dt_horizon_coef_00 = np.concatenate(
+                [raw_dt_horizon_coef_00, diagnostics_data[:, 8]]
+            )
+            smooth_damping_timescale = np.concatenate(
+                [smooth_damping_timescale, diagnostics_data[:, 9]]
+            )
+            min_delta_r = np.concatenate([min_delta_r, diagnostics_data[:, 10]])
+            min_relative_delta_r = np.concatenate(
+                [min_relative_delta_r, diagnostics_data[:, 11]]
+            )
+            avg_delta_r = np.concatenate([avg_delta_r, diagnostics_data[:, 12]])
+            avg_relative_delta_r = np.concatenate(
+                [avg_relative_delta_r, diagnostics_data[:, 13]]
+            )
+            target_char_speed = np.concatenate(
+                [target_char_speed, diagnostics_data[:, 15]]
+            )
+            min_char_speed = np.concatenate(
+                [min_char_speed, diagnostics_data[:, 16]]
+            )
+            min_comoving_char_speed = np.concatenate(
+                [min_comoving_char_speed, diagnostics_data[:, 17]]
+            )
+            char_speed_crossing_time = np.concatenate(
+                [char_speed_crossing_time, diagnostics_data[:, 18]]
+            )
+            comoving_char_speed_crossing_time = np.concatenate(
+                [comoving_char_speed_crossing_time, diagnostics_data[:, 19]]
+            )
+            delta_r_speed_crossing_time = np.concatenate(
+                [delta_r_speed_crossing_time, diagnostics_data[:, 20]]
+            )
+            damping_timescale = np.concatenate(
+                [damping_timescale, diagnostics_data[:, 22]]
+            )
+
+    # Apply stylesheet
+    if stylesheet is not None:
+        plt.style.use(stylesheet)
+
+    # Set up plots
+    fig, axes = plt.subplots(7, 1, sharex=True)
+    plt.rcParams["text.usetex"] = True
+
+    # Plot the selected quantities. Groupings are based both on relationship
+    # of quantities and also relative scales of quantities
+    axes = list(axes)
+    # Damping times are sort of the odd ones out. Put it with control error but
+    # on a different y-axis
+    top1 = axes[0].plot(times, control_error, label="Control Error")
```

Comment:
Does it make sense to plot `abs(control_error)` on a log scale?

Reply:
No, not really. I think we'd lose a lot of information about small oscillations that way.

Reply:
How so? You see small oscillations well on a log scale. I suggest trying a log scale, and/or plotting both. Anyway, this is up to you.